src/components/tables/EntityTable.tsx

```diff
@@ -10,51 +12,208 @@ import {
     TablePagination,
     TableRow,
     TableSortLabel,
+    TextField,
+    Toolbar,
     Tooltip,
+    Typography,
 } from '@mui/material';
+import { SupabaseClient } from '@supabase/supabase-js';
+import ExternalLink from 'components/shared/ExternalLink';
 import formatDistanceToNow from 'date-fns/formatDistanceToNow';
-import { MouseEvent, useState } from 'react';
+import { ChangeEvent, MouseEvent, useEffect, useState } from 'react';
 import { FormattedMessage, useIntl } from 'react-intl';
-import {
-    DeploymentStatus,
-    Entity,
-    EntityMetadata,
-} from 'stores/ChangeSetStore';
+import { TABLES } from 'services/supabase';
+import { PostgrestError, useClient } from 'supabase-swr';
+
+enum TableStatuses {
```

Comment:
There are a few types, interfaces, and enumerations defined in this file and I imagine more shall follow. I anticipate a file dedicated to entity table types may be worthwhile if the type definitions become unruly.